---
title: "BCB420 Assignment 1"
author: "Hyunwoo Kwon"
date: "2024/02/11"
output:
  html_document:
    toc: yes
    toc_depth: '2'
    df_print: paged
  html_notebook:
    toc: yes
    toc_depth: 2
---

## Introduction

The chose dataset comes from "Intracellular angiopoietin-1 promotes TKI-resistance via activation of JAK/STAT5 pathway in chronic myeloid leukemia" which was published in March, 2023 by Ma D et. al. The study aimed to investigate the role of angiopoietin-1(ANG-1) in drug resistance in chronic myeloid leukemia (CML), particulary its impact on resistance to BCR-ABL tyrosine kinase inhibitors and other chemotherapeutics. Researchers focused on understanding how overexpression of intracellular and secretory ANG-1 contributes to CML drug resistance, examining the effects of ANG-1 silencing in cell lines and animal models.

## Downloading the packages
```{r, message=FALSE}

if (!requireNamespace("BiocManager", quietly = FALSE)){
  install.packages("BiocManager")}

if (!requireNamespace("GEOmetadb", quietly = FALSE)){
  BiocManager::install("GEOmetadb")}

if (!requireNamespace("knitr", quietly = FALSE)){
  install.packages("knitr")}

if (!requireNamespace("edgeR", quietly = FALSE)){
  BiocManager::instal("edgeR")}

if (!requireNamespace("biomaRt", quietly = FALSE)){
  BiocManager::install("biomaRt")}

if (!requireNamespace("DBI", quietly = FALSE)){
  install.packages("DBI")}

if (!requireNamespace("GEOquery", quietly = FALSE)){
  BiocManager::instal("GEOquery")}

library(GEOquery)
library(knitr)
library(edgeR)
library(biomaRt)
library(dplyr)
library(purrr)

```

## Downloading the data

We use the GEOquery package (@R-GEOquery) to get the datasets.

```{r, message=FALSE}
geo_id <- "GSE236504"
#GEO dataset downloading
gse <- getGEO(geo_id, GSEMatrix = FALSE)
# Information about platform
current_gpl <- names(GPLList(gse))[1]
current_gpl_info <- Meta(getGEO(current_gpl))
gse@header$summary
```

**GEO ID:** `r geo_id`

**Dataset Name:** Transcription factor ATMIN facilitates chemoresistance in nasopharyngeal carcinoma

**Platform Title:** `r current_gpl_info$title`

**Technology:** `r current_gpl_info$technology`

**Submission Date:** `r current_gpl_info$submission_date`

**Last Update Date:** `r current_gpl_info$last_update_date`

**Organism:** `r current_gpl_info$organism`

**Number of GEO datasets that use this technology:** `length(current_gpl_info$series_id)`

**Number of GEO samples that use this technology:** `length(current_gpl_info$sample_id))`

Raw datasets are now loaded from the supplementary files.

```{r, message=FALSE}

# Downloads the data
#Get the name of the supplementary files
sfilenames = getGEOSuppFiles(geo_id, fetch_files = FALSE)
sfilenames$fname
#file to choose
data_filename <- sfilenames$fname[2] 
data_filename

#location to download and store files
# Save to current directory
dir_download <- file.path(getwd(), "Data")
missing_files <- sfilenames$fname[!unlist(
  lapply(sfilenames$fname, FUN=function(x){
    file.exists(
      file.path(dir_download,geo_id,x))}))]
if(length(missing_files) >0){
  for(i in 1:length(missing_files)){
    sfiles = getGEOSuppFiles(geo_id,
                             filter_regex = missing_files[i],
                             baseDir = dir_download,
                             fetch_files = TRUE)
  }
}

```

## Load and read the data

```{r, message=FALSE}
# Creating relative path for the downloaded dataset file
# Reading the data into a table
raw_data <- read.table(file.path(dir_download,geo_id,data_filename), header = TRUE, check.names = TRUE)
# Check if the dataset is complete through inspecting the dimensions.
dim(raw_data)

nrow(raw_data)

```

Now we can see that the dataset size is 54,261 genes with 11 rows. To see if column names:

```{r}
kableExtra::kable(raw_data[1:14,], format = "html")
colnames(raw_data)
```

### Filtering Rows with Low Counts

# Each column represent the 
# Further Investigation
```{r, message=FALSE}
gse@gsms[[1]]@header$characteristics_ch1
gse@gsms[[1]]@header$title
gse@gsms[[1]]@header$geo_accession
list_of_samples <- gse@gsms
samples_type <- do.call(rbind,
                       lapply(list_of_samples,
                              FUN=function(x){
                                c(x@header$title,
                                  x@header$characteristics_ch1)
                                  }
                              )
                      )
colnames(samples_type) <- c("title",
                           "cell line", "cell type","genotype")
samples_type[,"cell line"] <- gsub(samples_type['cell line'],
                                  pattern = "cell line: ",
                              replacement = "")
samples_type[,"cell type"] <- gsub(samples_type['cell type'],
                                  pattern = "cell type: ",
                              replacement = "")
samples_type[,"genotype"] <- gsub(samples_type['genotype'],
                                  pattern = "genotype: ",
                              replacement = "")

kable(samples_type[1:6,], format = "html")
```
# Converting to HUGO symbols
```{r}
summarized_gene_counts <- sort(table(raw_data$HUGO_symbol),
                               decreasing = TRUE)
kable(summarized_gene_counts[which(summarized_gene_counts>1)[1:10]],
      format = "html")
```
#Define the groups
```{r}
samples <- data.frame(lapply(colnames(raw_data)[2:6], 
                             FUN=function(x){unlist(strsplit(x, split = "kd"))}))

colnames(samples) <- colnames(raw_data)[2:6]
samples <- data.frame(t(samples))

## Clean the Data
```

```{r}

counts_name <- function(file_path, sample_name){
  df <- fread(file_path, select = c("geo_id", "counts"))
  df$sample <- sample_name
  return(df)
}

```


## Normalization
Now it is turn to normalize our data
```{r eval=TRUE}
numeric_data <- raw_data[, -1]  # Adjust as needed if there are multiple non-numeric columns

# Apply log2 transformation to numeric data only
data2plot <- log2(numeric_data + 1)  # Adding 1 to avoid log2(0)

boxplot(data2plot,xlab="Samples", ylab = "log2 CPM",
        las = 2, cex = 0.5, cex.lab = 0.5,
        cex.axis = 0.5, main = "RNASeq Samples")
abline(h = median(apply(data2plot, 2, median)),
       col = "green", lwd = 0.6, lty = "dashed")
```
```{r eval=TRUE}
par(mfrow=c(1,2))
#Density Plot for original count data
counts_density <- apply(log2(edgeR::cpm(raw_data[,2:7])), 2, density)

#Calculate the limits across all the samples
xlim <- 0; ylim <- 0
for (i in 1:length(counts_density)) {
  xlim <- range(c(xlim, counts_density[[i]]$x)); 
  ylim <- range(c(ylim, counts_density[[i]]$y))
}
cols <- rainbow(length(counts_density))
ltys <- rep(1, length(counts_density))
#plot the first density plot to initialize the plot
plot(counts_density[[1]], xlim=xlim, ylim=ylim, type="n", 
     ylab="Smoothing density of log2-CPM", main="Original Count", cex.lab = 0.8)
#plot each line
for (i in 1:length(counts_density)) lines(counts_density[[i]], col=cols[i])
#create legend
legend("topright", colnames(data2plot),  
       col=cols, lty=ltys, cex=0.5, 
       border ="blue",  text.col = "green4", 
       merge = TRUE, bg = "gray90")
```



What are the control and test conditions of the database?
The dataset compares bone marrow mononuclear cells from newly diagnosed or relapsed CML patients with and without drug treatment.

How many samples in each of the conditions of your datasets?
There are 14 samples in total, with replicates for both newly diagnosed and relapsed patients.

Were there expression values that were not unique for specific genes? How were you handle these?
In general, I was not able to spot the expression values that were not unique for specific genes. 

Were there expression values that could not be mapped to currernt HUGO symbols?

Were there any outliers in your dataset? How were they handled in the originating paper? How many outliers were removed?

How did you handle replicates?

What is the final coverage of your dataset?
